src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -74,15 +96,30 @@ export const TradeConfirm = ({ history }: RouterProps) => {
       }
     } catch (err) {
       console.error(`TradeConfirm:onSubmit - ${err}`)
-      // TODO: (ryankk) this needs to be revisited post bounty to handle actual errors coming back from unchained.
       let errorMessage
-      switch ((err as SwapError)?.message) {
-        case 'ZrxExecuteQuote - signAndBroadcastTransaction error: Error: Error signing & broadcasting tx': {
-          errorMessage = TRADE_ERRORS.TRANSACTION_REJECTED
+      switch ((err as ZrxError).message) {
```

Comment:
nicer to do this on line 97 rather than cast but this is just nit picking and fine here

Reply:
tsc doesn't allow that (rationale: https://github.com/Microsoft/TypeScript/issues/8677#issuecomment-220385124 😢)